File: backend/models/changelog.py
from typing import Literal

from pydantic import BaseModel


class ChangelogChange(BaseModel):
    type: Literal["added", "updated", "fixed", "removed"]
    category: str
    description: str


class ChangelogEntry(BaseModel):
    date: str
    version: str | None = None
    changes: list[ChangelogChange]
    last_updated: int | None = None

File: src/types/changelog.ts
export interface ChangelogChange {
  type: 'added' | 'updated' | 'fixed' | 'removed';
  category: string;
  description: string;
}

export interface ChangelogEntry {
  date: string;
  version?: string;
  changes: ChangelogChange[];
  last_updated?: number;
}
